src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -371,5 +372,56 @@ public CloudSdk build() {
           startListeners, runDevAppServerWaitSeconds);
     }
 
+    /**
+     * Attempt to find the Google Cloud SDK in various places.
+     * 
+     * @return the path to the root of the Google Cloud SDK
+     * @throws AppEngineException if not found
+     */
+    @Nonnull
+    private Path discoverSdkPath() {
+      for (CloudSdkResolver resolver : getResolvers()) {
+        try {
+          Path discoveredSdkPath = resolver.getCloudSdkPath();
+          if (discoveredSdkPath != null) {
+            return discoveredSdkPath;
+          }
+        } catch (Exception ex) {
```

Comment:
This is to prevent inadvertent exceptions from breaking the flow (e.g., `IOException` or `IllegalArgumentException`).  Perhaps I should just trap Throwable.